dali/operators/generic/constant_value.cc

```diff
@@ -0,0 +1,117 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/generic/constant_value.h"
+#include "dali/pipeline/operator/op_schema.h"
+#include "dali/core/convert.h"
+
+namespace dali {
+
+template <>
+void ConstantValue<CPUBackend>::RunImpl(Workspace &ws) {
+  auto &output = ws.Output<CPUBackend>(0);
+  const auto& out_shape = output.shape();
+  int nsamples = out_shape.size();
+  auto dtype = output.type();
+  auto &tp = ws.GetThreadPool();
+  if (has_fill_value_) {
+    auto &fill_value = ws.Input<CPUBackend>(kValueInputIdx);
+    const auto &fill_value_sh = fill_value.shape();
+    TYPE_SWITCH(fill_value.type(), type2id, FillValueType, (DALI_CONSTANT_VALUE_TYPES), (
+      TYPE_SWITCH(dtype, type2id, OutputType, (DALI_CONSTANT_VALUE_TYPES), (
+        for (int s = 0; s < nsamples; s++) {
+          tp.AddWork([&, s](int thread_idx) {
+            const auto* fill_value_data = fill_value.tensor<FillValueType>(s);
+            auto* out_data = output.mutable_tensor<OutputType>(s);
+            auto out_sz = out_shape.tensor_size(s);
+            auto fill_value_sz = fill_value_sh.tensor_size(s);
+            for (int64_t i = 0; i < out_sz; i++) {
+              out_data[i] = ConvertSat<OutputType>(fill_value_data[i % fill_value_sz]);
+            }
+          });
+        }
+        tp.RunAll();
+      ), (  // NOLINT
+        DALI_FAIL(make_string("Data type ", dtype, " is currently not supported. "
+                              "Supported types are : ", ListTypeNames<DALI_CONSTANT_VALUE_TYPES>()));
+      ));  // NOLINT
+    ), (  // NOLINT
+      DALI_FAIL(make_string("Data type ", fill_value.type(), " is currently not supported. "
+                            "Supported types are : ", ListTypeNames<DALI_CONSTANT_VALUE_TYPES>()));
+    ));  // NOLINT
+  } else {
+    TYPE_SWITCH(dtype, type2id, T, (DALI_CONSTANT_VALUE_TYPES), (
+      T value = ConvertSat<T>(const_value_);
+      for (int s = 0; s < nsamples; s++) {
+        tp.AddWork([&, value, s](int thread_idx) {
+          auto* out_data = output.mutable_tensor<T>(s);
+          auto out_sz = out_shape.tensor_size(s);
+          std::fill(out_data, out_data + out_sz, value);
+        });
+      }
+      tp.RunAll();
+    ), (  // NOLINT
+      DALI_FAIL(make_string("Data type ", dtype, " is currently not supported. "
+                            "Supported types are : ", ListTypeNames<DALI_CONSTANT_VALUE_TYPES>()));
+    ));  // NOLINT
+  }
+}
+
+DALI_SCHEMA(Full)
+    .DocStr(R"code(Returns new data of given shape and type, filled with a `fill_value`.)code")
```

Comment:
`fill_value` is not named argument. Can you update wording?

Reply:
done